### PR TITLE
Refacto quote

### PR DESCRIPTION
**Pull request by mbouchotoyst**

### CODE REVIEW

File: app/code/community/Oyst/OneClick/Model/Cart.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
+<?php
+
+/**
+ * Cart Model
+ */
+class Oyst_OneClick_Model_Cart
+{
+    public function initOystCheckout($params)
+    {
+        if (!empty($params['add_to_cart_form'])
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/Model/Cart.php]: For all if conditions, it's better when it's like that for future code modifications in code review
image

the modification impact will be only one line.

Same as **,** (comma) for the last element of array :)

File: app/code/community/Oyst/OneClick/Model/Catalog.php [hunk 2]
```diff
@@ -955,7 +956,8 @@ private function getCartRules(&$magentoQuoteBuilder, &$oneClickOrderCartEstimate
             $salesRules = $salesRuleCollection
                 ->addFieldToFilter('rule_id', array('in' => array_keys($total->getFullInfo())))
                 ->addFieldToFilter('simple_action', array('neq' => self::C4B_FREEPRODUCT_ADD_GIFT_ACTION))
-                ->setOrder('sort_order', $salesRuleCollection::SORT_ORDER_ASC);;
+                ->setOrder('sort_order', $salesRuleCollection::SORT_ORDER_ASC);
+            ;
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/Model/Catalog.php]: delete this ;)

File: app/code/community/Oyst/OneClick/Model/Magento/Quote.php [hunk 3]
```diff
@@ -236,7 +209,7 @@ private function getCustomer()
     /**
      * Consider all address info only from API same as New guest
      */
-    private function initializeAddresses()
+    private function syncDelivery()
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/Model/Magento/Quote.php]: method content also set billing address, so method name have to be changed

File: app/code/community/Oyst/OneClick/Model/OneClick/ApiWrapper.php [hunk 4]
```diff
@@ -63,15 +63,14 @@ public function authorizeOrder($dataFormated)
...
-            $dataFormated['isCheckoutCart'] = filter_var($dataFormated['isCheckoutCart'], FILTER_VALIDATE_BOOLEAN);
-
-            $this->getCartItems($dataFormated);
-        }
+        $this->getCartItems($dataFormated);
 
         /** @var Oyst_OneClick_Model_Catalog $oystCatalog */
         $oystCatalog = Mage::getModel('oyst_oneclick/catalog');
         $oystProducts = $oystCatalog->getOystProducts($dataFormated);
+        if (count($oystProducts) == 0) {
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/Model/OneClick/ApiWrapper.php]: why not ===

File: skin/frontend/base/default/js/oyst/oneclick.js [hunk 5]
```diff
@@ -261,21 +175,4 @@ function oneClickButtonPickToFirstAddToCartButtons(addtocartButtonsClass) {
             console.error(addtocartButtonsClass + " class not found.");
         }
     });
-}
-
-/**
- * Apply add to cart size on OneClick button
- */
-function smartButtonData(addtocartButtonsClass) {
```
Comment (mehdichaouch) [inline skin/frontend/base/default/js/oyst/oneclick.js]: Why you remove this ?

File: skin/frontend/base/default/js/oyst/oneclick.js [hunk 6]
```diff
@@ -16,88 +16,62 @@
 /**
  * Main function to init 1-Click
  *
- * @param {String} productTypeId Product Type Id
  * @param {String} oneClickUrl  Backend conf oneclick payment url
- * @param {Boolean} isProductAddtocartFormValidate  Backend conf to validate or not form data
  */
-function oystOneClick(productTypeId, oneClickUrl, isProductAddtocartFormValidate, addtocartButtonsClass) {
-    smartButtonData(addtocartButtonsClass);
```
Comment (mehdichaouch) [inline skin/frontend/base/default/js/oyst/oneclick.js]: Why this is missing ?

File: app/design/frontend/base/default/template/oyst/oneclick/checkout/cart/oneclick.phtml [hunk 7]
```diff
@@ -23,10 +23,12 @@
 <script>
     (function () {
         if ("function" === typeof oystOneClick) {
-            oystOneClick(
-                <?php echo $this->getOneClickUrl(); ?>,
-                <?php echo Mage::getSingleton('checkout/session')->getQuote()->getId(); ?>
-            );
+            var oystOneClickConfig = {
+                'oneClickUrl': <?php echo $this->getOneClickUrl(); ?>,
+                'isCheckoutCart': true,
```
Comment (mehdichaouch) [inline app/design/frontend/base/default/template/oyst/oneclick/checkout/cart/oneclick.phtml]: Checks JavaScript source code for last comma in object literal. Such object could have problem in certain JavaScript engine implementations.

File: app/code/community/Oyst/OneClick/Model/Magento/Quote.php [hunk 8]
```diff
@@ -86,9 +73,9 @@ public function buildQuote()
     /**
      * @param int $storeId
      */
-    private function initializeQuote($storeId = null)
+    private function syncQuote($quoteId, $storeId = null)
     {
-        $this->quote = Mage::getModel('sales/quote');
+        $this->quote = Mage::getModel('sales/quote')->load($quoteId);
```
[the remark this answers is not in the document]
Comment (mbouchotoyst) [reply app/code/community/Oyst/OneClick/Model/Magento/Quote.php]: This call is done once, so Singleton or Model makes no difference.

**Pull request closed without merging**